File: BankingTool.Model/Dto/RequestDtos/CreateTokenRequest.cs
namespace BankingTool.Model.Dto.RequestDtos
{
    public class CreateTokenRequestObject
    {
        public CreateTokenRequest Request { get; set; }
    }
    public class CreateTokenRequest
    {
        public CreateTokenRequestUser User { get; set; }
        public CreateTokenRequestRole Role { get; set; }
    }
    public class CreateTokenRequestUser : RequestId
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
    }
    public class CreateTokenRequestRole : RequestId
    {
    }
}

File: bankingtool.client/src/app/dto/request/user/create-token-request.ts
import { RequestId } from "../request-id";

export interface CreateTokenRequestObject {
  request: CreateTokenRequest;
}

export interface CreateTokenRequest {
  user: CreateTokenRequestUser;
  role: CreateTokenRequestRole;
}

export interface CreateTokenRequestUser extends RequestId {
  firstName: string;
  lastName: string;
  email: string;
}

export interface CreateTokenRequestRole extends RequestId {

}
